zone/bot_database.cpp

```diff
@@ -379,86 +380,47 @@ bool BotDatabase::LoadBot(const uint32 bot_id, Bot*& loaded_bot)
 		return false;
 	}
 
-	query = fmt::format(
-		"SELECT"
-		" `owner_id`," // 0
-		" `spells_id`," // 1
-		" `name`," // 2
-		" `last_name`," // 3
-		" `title`," // 4
-		" `suffix`," // 5
-		" `level`," // 6
-		" `race`," // 7
-		" `class`," // 8
-		" `gender`," // 9
-		" `size`," // 10
-		" `face`," // 11
-		" `hair_style`," // 12
-		" `hair_color`," // 13
-		" `eye_color_1`," // 14
-		" `eye_color_2`," // 15
-		" `beard`," // 16
-		" `beard_color`," // 17
-		" `drakkin_heritage`," // 18
-		" `drakkin_tattoo`," // 19
-		" `drakkin_details`," // 20
-		" `hp`," // 21
-		" `mana`," // 22
-		" `time_spawned`," // 23
-		" `zone_id`," // 24
-		" `show_helm`," // 25
-		" `follow_distance`," // 26
-		" `stop_melee_level`," // 27
-		" `expansion_bitmask`," // 28
-		" `enforce_spell_settings`," // 29
-		" `archery_setting`" // 30
-		" FROM `bot_data`"
-		" WHERE `bot_id` = {}"
-		" LIMIT 1",
-		bot_id
+	const auto& l = BotDataRepository::GetWhere(
+		database,
+		fmt::format(
+			"bot_id = {}",
+			bot_id
+		)
 	);
-
-	auto results = database.QueryDatabase(query);
-	if (!results.Success()) {
+	if (l.empty()) {
 		return false;
 	}
 
-	if (!results.RowCount()) {
-		return true;
-	}
-
-	auto row = results.begin();
-
 	auto d = Bot::CreateDefaultNPCTypeStructForBot(
-		std::string(row[2]), // Name
-		std::string(row[3]), // Last Name
-		static_cast<uint8>(EQ::Clamp(std::stoi(row[6]), 1, 255)), // Level
-		static_cast<uint16>(std::stoul(row[8])), // Race
-		static_cast<uint8>(EQ::Clamp(std::stoi(row[9]), WARRIOR, BERSERKER)), // Class
-		static_cast<uint8>(EQ::Clamp(std::stoi(row[7]), MALE, FEMALE)) // Gender
+		l[0].name,
+		l[0].last_name,
+		l[0].level,
+		l[0].race,
+		l[0].class_,
+		l[0].gender
 	);
 
 	auto t = Bot::FillNPCTypeStruct(
-		std::stoul(row[1]), // Spells ID
-		std::string(row[2]), // Name
-		std::string(row[3]), // Last Name
-		static_cast<uint8>(EQ::Clamp(std::stoi(row[6]), 1, 255)), // Level
-		static_cast<uint16>(std::stoul(row[7])), // Race
-		static_cast<uint8>(EQ::Clamp(std::stoi(row[8]), WARRIOR, BERSERKER)), // Class
-		static_cast<uint8>(EQ::Clamp(std::stoi(row[9]), MALE, FEMALE)), // Gender
-		std::stof(row[10]), // Size
-		std::stoul(row[11]), // Face
-		std::stoul(row[12]), // Hair Style
-		std::stoul(row[13]), // Hair Color
-		std::stoul(row[14]), // Eye Color 1
-		std::stoul(row[15]), // Eye Color 2
-		std::stoul(row[16]), // Beard
-		std::stoul(row[17]), // Beard Color
-		std::stoul(row[18]), // Drakkin Heritage
-		std::stoul(row[19]), // Drakkin Tattoo
-		std::stoul(row[20]), // Drakkin Details
-		std::stoi(row[21]), // Health
-		std::stoi(row[22]), // Mana
+		l[0].spells_id,
```

Comment:
Would really prefer we use the `FindOne` method when we intend on working with one object

It looks like the repository template will need to be updated to use the primary key in the `fmt::format`

```cpp
	static BotData FindOne(
		Database& db,
		int bot_data_id
	)
	{
		auto results = db.QueryDatabase(
			fmt::format(
				"{} WHERE id = {} LIMIT 1",
				BaseSelect(),
				bot_data_id
			)
		);
```

Reply:
Then you are accessing just the one object via `l->` versus `l[0]->` which is a bit strange to read

Reply:
Ah yeah, forgot about FindOne while I was working on this. No worries.